Создал поиск, добавил шрифты, также был добавлен функционал для экран…

Nikita-Gavrilov-JK opened the pull request:
Сделал задание 9 спринта 
Реализовал тёмную тему в настройках и др. функционал тоже
Экран Поиска:  при повороте экрана текст сохраняется,
кнопка "Очистить" стирает текст и убирает клавиатуру

Nikita-Gavrilov-JK committed:
Создал поиск, добавил шрифты, также был добавлен функционал для экрана "Настройки"

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,51 @@
+        val backButSearch = findViewById<ImageButton>(R.id.backButtonSearch)
+        backButSearch.setOnClickListener {
+            val backButIntent = Intent(this, MainActivity::class.java)
+            startActivity(backButIntent)
```
Для перехода назад надо использовать не интент, а методы Activity finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на этот экран. На эту тему можно почитать про стек активити.

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,51 @@
+            }
+
+            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+                clearButton.visibility = if (s.isNotEmpty()) View.VISIBLE else View.GONE
```
Для изменения видимости элемента на экране в данном случае удобнее использовать extension View.isVisible:
View.isVisible = true соответствует View.visibility = View.VISIBLE
View.isVisible = false соответствует View.visibility = View.GONE

faritowich commented on app/src/main/res/layout/activity_main.xml:
dark_activity_settings.xml и dark_activity_main надо удалить из проекта

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -16,5 +20,49 @@ class SettingsActivity : AppCompatActivity() {
+            }
+
+            val shareIntent = Intent.createChooser(sendIntent, null)
+            startActivity(shareIntent)
```
Можно упростить до
```
Intent().apply {
    action = Intent.ACTION_SEND
    putExtra(Intent.EXTRA_TEXT, getString(R.string.link_android_practicum))
    type = "text/plain"
    startActivity(Intent.createChooser(this, null))
}
```

Nikita-Gavrilov-JK committed:
Исправил переход назад на экранах "Поиск" и "Настройки"
убрал метод Intent вместо него добавил метод finish

Nikita-Gavrilov-JK commented:
Среди замечаний было ещё 2
При нажатии на "Написать в поддержку" у меня появляется выбор приложения я выбрал gmail почту (фото присылаю)
Так же было замечание насчёт того, что текст не сохраняется при повороте у меня текст сохраняется
Проверял на Honor 10i
![Screenshot1 app](https://github.com/Nikita-Gavrilov-JK/PlaylistMaker/assets/102280514/3e079eae-8944-47d6-a230-182d58b0354f)
![Screenshot2 app](https://github.com/Nikita-Gavrilov-JK/PlaylistMaker/assets/102280514/6fd7853c-429b-4e46-80e7-bd42e27f7d5e)
![Screenshot4](https://github.com/Nikita-Gavrilov-JK/PlaylistMaker/assets/102280514/995a6e50-5465-4b5a-92c4-090952005cb0)
![Screenshot3](https://github.com/Nikita-Gavrilov-JK/PlaylistMaker/assets/102280514/920629f5-0c4f-4e4d-8bb3-41733f7ad014)

Nikita-Gavrilov-JK committed:
Исправил нажатие кнопки "Написать в поддержку"
Были переопределены методы onSaveInstantState и onRestoreInstantState для сохранения текста в поле поиска

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,64 @@
+        }
+
+        val editText = findViewById<EditText>(R.id.edit_text)
+        editText.setText(searchText)
```
Надо сеттить или тут, или в onRestoreInstanceState. В обоих местах сразу лишнее

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,64 @@
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        searchText = savedInstanceState.getString("searchText", "")
```
"searchText" и вообще строки-ключи стоит выносить в отдельную константу (в данном случае в companion object). Это позволит не бояться забыть поменять в одном месте и забыть в другом. При необходимости просто меняешь в самой константе и всё так же работает. Плюс сразу понятно, где мы передаем и получаем эти данные.

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,64 @@
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        searchText = savedInstanceState.getString("searchText", "")
+        val editText = findViewById<EditText>(R.id.edit_text)
```
Сейчас для вью R.id.edit_text переменная объявляется дважды в разных методах. Этого можно избежать, если объявлять глобальную переменную в теле класса с lateinit (будет private lateinit var viewName: EditText) или как нуллабл переменную (будет private var viewName: EditText? = null, а в onCreate уже задавать значение findViewById<EditText>(R.id.VIEW_ID). Все остальные переменные, отвечающие за вьюхи, можно объявить так же. Тогда переменную можно инициализировать один раз в onCreate и иметь к ней доступ из любой функции класса.
